### ISSUE TITLE
ValueError: zero-size array to reduction operation minimum which has no identity

### ISSUE THREAD

**Issue by yangke** [a0x77n/chucky-ng#11]
Also running on the pidgin-2.10.0 codebase
when downsize the neighbors to 10

```
python  chucky.py -n 10 -i callee  atoi
```

```
python chucky.py -n 10 -i callee atoi
Job (1/140): jabber_ibb_session_create_from_xmlnode (1359885) - jabber_ibb_session_create_from_xmlnode atoi [Callee]
 0.14962    jabber_ibb_session_create_from_xmlnode         1359885  $RET
Job (2/140): gntwm_init (69588) - gntwm_init atoi [Callee]
 0.21480    gntwm_init                                       69588  $ARG
Job (3/140): msn_parse_addressbook_contacts (1080546) - msn_parse_addressbook_contacts atoi [Callee]
...
Job (21/140): google_session_handle_candidates (1371190) - google_session_handle_candidates atoi [Callee]
 0.12511    google_session_handle_candidates               1371190  $RET
Job (22/140): im_mime_convert (1403094) - im_mime_convert atoi [Callee]
 0.14681    im_mime_convert                                1403094  NULL
Job (23/140): ipg_cmd (1014135) - ipg_cmd atoi [Callee]
Traceback (most recent call last):
  File "../python/anomaly_score.py", line 60, in <module>
    tool.run()
  File "/usr/local/lib/python2.7/dist-packages/joerntools-0.1-py2.7.egg/joerntools/shelltool/PipeTool.py", line 37, in run
    self._processStream()
  File "/usr/local/lib/python2.7/dist-packages/joerntools-0.1-py2.7.egg/joerntools/shelltool/PipeTool.py", line 49, in _processStream
    self.streamStart()
  File "../python/anomaly_score.py", line 33, in streamStart
    self.emb = self._loadEmbedding(self.args.dirname)
  File "../python/anomaly_score.py", line 26, in _loadEmbedding
    return self.loader.load(dirname)
  File "/usr/local/lib/python2.7/dist-packages/joerntools-0.1-py2.7.egg/joerntools/mlutils/EmbeddingLoader.py", line 20, in load
    self.emb.x, self.emb.y = load_svmlight_file(dirname + EMBEDDING_FILENAME)
  File "/usr/local/lib/python2.7/dist-packages/sklearn/datasets/svmlight_format.py", line 113, in load_svmlight_file
    zero_based, query_id))
  File "/usr/local/lib/python2.7/dist-packages/sklearn/datasets/svmlight_format.py", line 237, in load_svmlight_files
    or zero_based == "auto" and all(np.min(tmp[1]) > 0 for tmp in r)):
  File "/usr/local/lib/python2.7/dist-packages/sklearn/datasets/svmlight_format.py", line 237, in <genexpr>
    or zero_based == "auto" and all(np.min(tmp[1]) > 0 for tmp in r)):
  File "/usr/local/lib/python2.7/dist-packages/numpy/core/fromnumeric.py", line 2214, in amin
    out=out, keepdims=keepdims)
  File "/usr/local/lib/python2.7/dist-packages/numpy/core/_methods.py", line 21, in _amin
    out=out, keepdims=keepdims)
ValueError: zero-size array to reduction operation minimum which has no identity
[ERROR] Command 'echo 1014135 |python ../python/anomaly_score.py -d .chucky/tmpcOX8Bs/exp' returned non-zero exit status 1
[ERROR] Do not clean up.
Job (24/140): iro_cmd (1094745) - iro_cmd atoi [Callee]
Traceback (most recent call last):
  File "../python/anomaly_score.py", line 60, in <module>
    tool.run()
  File "/usr/local/lib/python2.7/dist-packages/joerntools-0.1-py2.7.egg/joerntools/shelltool/PipeTool.py", line 37, in run
    self._processStream()
  File "/usr/local/lib/python2.7/dist-packages/joerntools-0.1-py2.7.egg/joerntools/shelltool/PipeTool.py", line 49, in _processStream
    self.streamStart()
  File "../python/anomaly_score.py", line 33, in streamStart
    self.emb = self._loadEmbedding(self.args.dirname)
  File "../python/anomaly_score.py", line 26, in _loadEmbedding
    return self.loader.load(dirname)
  File "/usr/local/lib/python2.7/dist-packages/joerntools-0.1-py2.7.egg/joerntools/mlutils/EmbeddingLoader.py", line 20, in load
    self.emb.x, self.emb.y = load_svmlight_file(dirname + EMBEDDING_FILENAME)
  File "/usr/local/lib/python2.7/dist-packages/sklearn/datasets/svmlight_format.py", line 113, in load_svmlight_file
    zero_based, query_id))
  File "/usr/local/lib/python2.7/dist-packages/sklearn/datasets/svmlight_format.py", line 237, in load_svmlight_files
    or zero_based == "auto" and all(np.min(tmp[1]) > 0 for tmp in r)):
  File "/usr/local/lib/python2.7/dist-packages/sklearn/datasets/svmlight_format.py", line 237, in <genexpr>
    or zero_based == "auto" and all(np.min(tmp[1]) > 0 for tmp in r)):
  File "/usr/local/lib/python2.7/dist-packages/numpy/core/fromnumeric.py", line 2214, in amin
    out=out, keepdims=keepdims)
  File "/usr/local/lib/python2.7/dist-packages/numpy/core/_methods.py", line 21, in _amin
    out=out, keepdims=keepdims)
ValueError: zero-size array to reduction operation minimum which has no identity
[ERROR] Command 'echo 1094745 |python ../python/anomaly_score.py -d .chucky/tmphNLAZb/exp' returned non-zero exit status 1
[ERROR] Do not clean up.
Job (25/140): parse_priority (1358843) - parse_priority atoi [Callee]
 0.09366    parse_priority                                 1358843  NULL
Job (26/140): gg_watch_fd (1588677) - gg_watch_fd atoi [Callee]
 0.18309    gg_watch_fd                                    1588677  $RET
...
```

**Comment by fabsx00** [comment]
Try removing the .chucky-directory. This error occurs when ALL condition vectors are null-vectors. We can catch this at some point but I think the reason you're seeing this is because there's an old cache lying around. Please also note that caching of condition-vectors is NOT valid as condition vectors depend on the symbol that is being analyzed.

**Comment by yangke** [comment]
Hi fabs,
Thank you for reminding me the error code!
I've closed the pull request.

For this "Value Error" issue,I have tried to make sure the chucky-directory is cleaned.
I removed all the code  and reclone  the alwin's latest version.
This problem still exists.
It seems not the problem of uncleaned directory ".chucky" 

And the second,I'm now aware of your caching suggestion.
I just considered the command I tested and forgot there are also 
command using functions and other symbols where the jobs iteratively executed doesn't use the same symbol  for condition  tainting.I'll modify it latter.

But it's necessary to cache the condition,when doing the work like 
python chucky.py -n 25 -i callee atoi.
Because the taint symbol is fixed to "atoi".
And I find it's  apparently faster when most of the function condition are cached.
at the back   leaning stage of ,for example,140 jobs in pidgin-2.10.0 code database.
